src/main/java/org/kitteh/irc/client/library/implementation/NettyManager.java

```diff
@@ -264,12 +266,27 @@ private void schedule(boolean force) {
                 if (this.scheduledPing != null) {
                     this.scheduledPing.cancel(false);
                 }
-                this.scheduledSending = this.channel.eventLoop().scheduleAtFixedRate(() -> {
+                if (this.zeroDelayFuture != null) {
+                    this.zeroDelayFuture.cancel(false);
+                }
+                final Runnable pollSendRunnable = () -> {
                     String message = ClientConnection.this.queue.poll();
                     if (message != null) {
                         ClientConnection.this.channel.writeAndFlush(message);
                     }
-                }, delay, this.client.getMessageDelay(), TimeUnit.MILLISECONDS);
+                };
+                if (this.client.getMessageDelay() == 0) {
+                    final Runnable zeroDelayRunnable = new Runnable() {
+                        @Override
+                        public void run() {
+                            pollSendRunnable.run();
+                            ClientConnection.this.zeroDelayFuture = ClientConnection.this.channel.eventLoop().submit(this);
+                        }
+                    };
+                    zeroDelayFuture = this.channel.eventLoop().submit(zeroDelayRunnable);
```

Comment:
Minor issues like this don't matter if there are fundamental problems with the approach